lesson_10/homeworks.py

```diff
@@ -1,2 +1,75 @@
-def func():
-    pass
\ No newline at end of file
+
+'''Рахування унікальних символів в строці
+ Порахувати кількість унікальних символів в строці. Якщо їх більше 10 - вивести в консоль True, 
+ інакше - False. Строку отримати за допомогою функції input()'''
+ 
+def count_unique_characters(str_for_count:str) -> bool:
+    """Functions count unique characters in string and:\n
+    - returns 'True' if string contains more than 10 unique characters;\n
+    - returns 'False' if string contains less than 10 unique characters.
+    """
+    if not isinstance (str_for_count, str):
+        raise TypeError('Only \'str\' type is expected.')
+    set_for_count = set(str_for_count)
+    set_len = len(set_for_count)
+    if set_len == 0:
+        raise ValueError('Empty string is entered.')
+    if set_len > 10:
+        return(True)
+    if set_len <= 10:
+        return(False)
+  
+""" Написати функцію, яка приймає два рядки та повертає індекс першого входження другого рядка
+у перший рядок, якщо другий рядок є підрядком першого рядка, та -1, якщо другий рядок
+не є підрядком першого рядка."""
+
+def find_substring(str1:str, str2:str) -> int:
+    """ Function checks if str1 contains str2.\n 
+    Returns index (position) of the first occurrence of str2 in str1.\n
+    Returns '-1' if str1 does not contain str2."""
+    if not isinstance (str1, str):
+        raise TypeError('Only \'str\' type is for str1.')
+    if not isinstance (str2, str):
+        raise TypeError('Only \'str\' type is for str2.')
+    if len(str1) < len(str2):
+        raise ValueError('str1 is expected to be longer than str2.')
+    if str2 in str1:
+        return(str1.index(str2))
+    else:
+        return(-1)
```

Comment:
```suggestion
    return str1.index(str2)
```
і воно поверне -1 або індекс

Reply:
ось тут я не дуже зрозуміла як саме потрібно виправити функцію? Тобто код у рядках 36-39 замінити на return str1.index(str2)? Тоді фінальний вигляд функції :

```
def find_substring(str1:str, str2:str) -> int:
    """ Function checks if str1 contains str2.\n 
    Returns index (position) of the first occurrence of str2 in str1.\n
    Returns '-1' if str1 does not contain str2."""
    if not isinstance (str1, str) or not isinstance (str2, str):
        raise TypeError('Only \'str\' type is supported for both values.')
    if len(str2) == 0:
        raise ValueError('str2 is empty.')
    return str1.index(str2)
```

Для такої функції отримую еррору:

```
Traceback (most recent call last):
  File "c:\Users\iuliia.lazurkevych\aqa_070225\lesson_10\homeworks.py", line 71, in <module>
    print(f'Func 2: Index of the first occurrence of str2 in str1 (-1 if str1 does not contain str2): {find_substring(str1, str2)}') # поверне -1
                                                                                                       ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\iuliia.lazurkevych\aqa_070225\lesson_10\homeworks.py", line 35, in find_substring
    return str1.index(str2)
           ^^^^^^^^^^^^^^^^
ValueError: substring not found
```

Reply:
я помилився, виправив